portal/src/texts/blog/six-advice-on-mixpanel.mdx

```diff
@@ -0,0 +1,68 @@
+</Ingress>
+
+User analytics is one of the most fundamental features that every popular web application should perform.
+Without collecting data about user activities, one can never be sure of the user experience.
+The developer team will invest days on some functionality, but unless they know how many users use this precious feature,
+they will only guess and hope that the users are benefiting from it.
+In other words, “Without data you’re just another person with an opinion.” W. Edwards Deming.
+
+![Edwards Deming](/assets/blog/quote-edwards-deming-data-opinion.jpg)
+
+To clarify the ambiguity, my production team of [https://meldeskade.no/](https://meldeskade.no) decided to use popular business analytics tool, [Mixpanel](https://mixpanel.com/).
+Our single page web application is built with ReactJS, uses Jøkul Design System and hundreds of users visit the website
+to report their insurance activities every day.
+In short, meldskade.no is a typical SPA which promises the user to follow a set of operations
+under some sections and complete the user request somewhere along the way.
+
+From the perspective of a front-end developer, I thought including Mixpanel tracking would be a simple task. However, as our team gained more and more experience in the field, I started to realize the obstacles. We followed the path of ‘learning by doing’ and eventually, our Mixpanel tracking started to deliver us data. So, this article will help to those who are in the beginning of their Mixpanel journey and looking for some useful advice.
+
+### #1 Don’t underestimate the integration work
+
+Know your enemy first! Adding Mixpanel to your web application is an integration process and system integration is maybe the most problematic fields in information technology. Because linking different systems increases the complexity. Although the relationship between a web application and Mixpanel is a one-way road, the integration will still be painful, and it has the potential to produce unforeseen errors.
+
+For instance, if you perform some tracking event before initializing the Mixpanel with the code below, application will have a fatal error and whole website will be down for the user. This error can seem obvious, but one exceptional case for some users might be the end of the show. Therefore, do not underestimate the complexity of integration process.
+
+```javascript
+import { MixpanelUtils } from './MixpanelUtils';
+...
+MixpanelUtils.Init();
+```
+
+### #2 Document every tracking operation first, code later
+
+Teamwork is required in order to integrate a web application with Mixpanel. Naturally, a web developer cannot sit alone and write all the tracking operations on his/her own for a large system. Product owner, architect, tester, and tech lead should give their consent on what user activities will be tracked besides developers. Since people from different roles are involved in the process, it is important to keep decisions in one open document.
```

Comment:
En mulighet i stedet for his/her er å bruke their.

```suggestion
Teamwork is required in order to integrate a web application with Mixpanel. Naturally, a web developer cannot sit alone and write all the tracking operations on their own for a large system. Product owner, architect, tester, and tech lead should give their consent on what user activities will be tracked besides developers. Since people from different roles are involved in the process, it is important to keep decisions in one open document.
```